src/fsharp/service/service.fs

```diff
@@ -1313,6 +1313,18 @@ type TypeCheckInfo
         let isDisposableTy (ty: TType) =
             protectAssemblyExplorationNoReraise false false (fun () -> Infos.ExistsHeadTypeInEntireHierarchy g amap range0 ty g.tcref_System_IDisposable)
 
+        let isStructTyconRef (tyconRef: TyconRef) =
```

Comment:
You should be able to just use `isStructTy g (generalizeTyconRef tcref)`.

Note that
* `TypeAbbrev` should never really be expanded manually, you should always use helpers like `isStructTy` that go through `stripTyEqns`.  This will also cope with equations from type inference

* It's normally best not to do `CanDeref` manually, that should always be looked after in helpers.  You should be operating inside a protect-assembly-exploration area which will cope with an exception from incomplete assembly sets